### PR TITLE
Auto update channel default

### PR DESCRIPTION
**Pull request by reset**
This will resolve an issue where the new `--auto-update` flag was
incorrectly automatically retrieving updates from the unstable
channel by default. This change should additionally prevent mistakes
like this from escaping in the future by explicitly requiring a channel
and more clearly stating how our default channel is calculated.

tenor-215320992

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/sup/src/main.rs [hunk 1]
```diff
@@ -163,7 +164,7 @@ fn cli<'a, 'b>() -> App<'a, 'b> {
             (@arg ENVIRONMENT: --environment -e +takes_value requires[APPLICATION]
                 "Environment name; [default: not set].")
             (@arg CHANNEL: --channel +takes_value
-                "Receive package updates from the specified release channel")
+                "Receive package updates from the specified release channel [default: stable]")
```
Comment (fnichol) [inline components/sup/src/main.rs]: 👍

File: components/sup/src/manager/mod.rs [hunk 2]
```diff
@@ -116,7 +116,7 @@ pub struct ManagerConfig {
     pub auto_update: bool,
     pub eventsrv_group: Option<ServiceGroup>,
     pub update_url: String,
-    pub update_channel: Option<String>,
+    pub update_channel: String,
```
Comment (fnichol) [inline components/sup/src/manager/mod.rs]: Yeah, I like this better--forces us to determine the defaults upstream and keeps the internals simpler.

File: components/sup/src/main.rs (hunk 1 above)
Comment (reset) [inline components/sup/src/main.rs]: I'm pretty sure this was an intermediary step in the first place before we had channels pipe through in the UI of Builder just so we could avoid confusion

**Comment by reset** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/257121625).

**Comment by thesentinels** [comment]
:broken_heart: Travis CI reports [this PR failed to pass the test suite](https://travis-ci.org/habitat-sh/habitat/builds/257121625).

The next step is to examine the [job](https://travis-ci.org/habitat-sh/habitat/builds/257121625) and figure out why. If it is transient, you can try re-triggering the Travis CI Job - if it passes, this PR will be automatically merged. If it is not transient, you should fix the issue and update this pull request, and issue `approve` again. If you believe it will never pass, and you are feeling :godmode:, you can issue a `force` to merge this PR anyway.

**Pull request merged**